src/main/java/com/iexec/core/task/TaskService.java

```diff
@@ -432,6 +434,63 @@ private void initializedOrRunning2ContributionTimeout(Task task) {
         }
     }
 
+    /**
+     * When all workers have failed to run the task, we send the new status as soon as possible.
+     * It avoids a state within which we know the task won't be updated anymore, but we wait for a timeout.
+     * <br>
+     * We consider that all workers are in a `RUNNING_FAILED` status when:
+     * <ol>
+     *     <li>All alive workers have tried to run the task;</li>
+     *     <li>All workers that have tried to run the task have failed, whether they are alive or lost;</li>
+     *     <li>No replicate has reached the `COMPUTED` status.</li>
+     * </ol>
+     *
+     * @param task Task to check and to make become {@link TaskStatus#RUNNING_FAILED}.
+     */
+    private void running2RunningFailed(Task task) {
+        boolean isRunningTask = task.getCurrentStatus().equals(RUNNING);
+        if (!isRunningTask) {
+            return;
+        }
+
+        final List<Replicate> replicates = replicatesService.getReplicates(task.getChainTaskId());
+
+        final List<Worker> aliveWorkers = workerService.getAliveWorkers();
+
+        // If an alive worker has not run the task, it is not a `RUNNING_FAILURE`.
+        final Predicate<Worker> hasWorkerAlreadyTried = worker -> replicates
+                .stream()
+                .map(Replicate::getWalletAddress)
+                .anyMatch(address -> address.equals(worker.getWalletAddress()));
+
+        final boolean allAliveWorkersTried = aliveWorkers
+                .stream()
+                .allMatch(hasWorkerAlreadyTried);
+        if (!allAliveWorkersTried) {
+            return;
+        }
+
+        // If all workers that have run the task have failed while running the task, it is definitely a `RUNNING_FAILURE`.
+        boolean allReplicatesFailed = replicates
+                .stream()
+                .map(Replicate::getLastRelevantStatus)
+                // A replicate that has not ever been created nor started is probably failed.
```

Comment:
I'm absolutely not sure about that. I've based my work on the `getLastRelevantStatus` return type. That could be wrong.

However, `Replicate` provides a no-args constructor which does not initialize `statusUpdateList`. This constructor seems to be used only in test cases so not a problem as of now... But that could be an issue in the future. 
We could do the following:
1. Make `getLastRelevantStatus` return a `ReplicateStatus` instead of an `Optional<ReplicateStatus>`;
2. Remove the `@NoArgsConstructor` annotation of `Replicate`;
3. Create a new no-args constructor with a single line to initialize the `statusUpdateList` as it is already done in the other constructor: `this.statusUpdateList.add(poolManagerRequest(CREATED));`.

If it's OK for you, I'll have a look at it in another PR.